relayer/src/foreign_client.rs

```diff
@@ -277,6 +279,46 @@ impl HasExpiredOrFrozenError for ForeignClientError {
     }
 }
 
+/// Optional onfiguration parameters for the CreateClient command.
+/// If set, the options override the defaults taken from the chain configuration.
+#[derive(Debug, Default)]
+pub struct CreateOptions {
+    /// The `max_clock_drift` value for a client state. If set to none,
+    /// The default value is computed as a function
+    /// of the configurations of the source chain and the destination chain.
+    ///
+    /// The client state clock drift must account for destination
+    /// chain block frequency and clock drift on source and dest.
+    /// https://github.com/informalsystems/ibc-rs/issues/1445
+    pub max_clock_drift: Option<Duration>,
+
+    pub trusting_period: Option<Duration>,
+    pub trust_threshold: Option<TrustThreshold>,
+}
+
+impl CreateOptions {
+    fn build_client_settings(
+        self,
+        src_chain_config: &ChainConfig,
+        dst_chain_config: &ChainConfig,
+    ) -> ClientSettings {
+        let max_clock_drift = self.max_clock_drift.unwrap_or_else(|| {
+            src_chain_config.clock_drift
```

Comment:
Can we keep `calculate_client_state_drift` and use it here to make the code a bit more self-documenting?

Reply:
imho we should keep as much of this in `cosmos.rs`. While `max_clock_drift` is a chain parameter and expected to be configured for all chains, the `trusting_period` and `trust_threshold` are tendermint specific and we should avoid having them in clear in this file. This problem exists in master and not introduced by this PR but we shouldn't make things worse. Ideally we should check in the CLI that the parameters apply for the source chain type and in this file `CreateOptions` should be serialized, later decoded in `cosmos.rs`. 

Here is the client state for the solomachine:
https://github.com/informalsystems/ibc-rs/blob/06cff5763459dd8ebf054131cc146a1b9a441358/proto/src/prost/ibc.lightclients.solomachine.v2.rs#L4-L17

Reply:
I agree @ancazamfir, but I think `CreateOptions` could be an enum, with `CreateOptions::Tendermint` as the only currently defined variant carrying the tendermint-specific options.